packages/components/Callout/macos/CalloutView.swift

```diff
@@ -174,15 +226,138 @@ class CalloutView: RCTView, CalloutWindowLifeCycleDelegate {
 
 		isCalloutWindowShown = false
 	}
+	
+	// Return the TextView and TextShadowView of a Leaf node ShadowView for specialized nested TextView anchoring
+	private func getTextViewsForLeafShadow(leafShadowView: RCTShadowView) -> (textView: RCTTextView, textShadowView: RCTTextShadowView)? {
+		// Do not proceed if the preconditions of this function are not met
+		guard leafShadowView.isYogaLeafNode() else {
+				preconditionFailure("leafshadow is not a leaf node")
+		}
+
+		// Do not proceed if we don't have a UI Manager; we won't be able to determine the bounding rect
+		guard let uiManager = bridge?.uiManager else {
+				return nil
+		}
+
+		// Do not proceed if the leaf shadow is not a root text shadow view, which we're specializing for right now
+		// In the future this could be generalized for other complex yoga leaf nodes with subviews, such as react-native-svg
+		guard let textShadowView = leafShadowView as? RCTTextShadowView else {
+				return nil
+		}
+
+		// Do not proceed if we don't have the NSView corresponding to the yoga leaf view
+		guard let leafView = uiManager.view(forReactTag: leafShadowView.reactTag) else {
+				return nil
+		}
+
+		// Do not proceed if somehow the leafView is not an RCTTextView
+		guard let textView = leafView as? RCTTextView else {
+				preconditionFailure("it should not be possible for the RCTTextShadowView to not be represented by an RCTTextView")
+		}
```

Comment:
nit: We should be able to combine a number of these, at the very least the first and third ones. You could even chain it with the fourth one if you wanted unless you want to isolate that precondition failure case.

```swift
guard let leafView = bridge?.uiManager?.view(forReactTag: leafShadowView.reactTag) else {
    return nil
}
```